### DISCUSSION TITLE
Many clientWebsockets working in one thread

### DISCUSSION THREAD

**Discussion by HACKONDEX** [ithewei/libhv#288] [Q&A] [answered]
I want to use WebsocketClient and I need 10 connections at the same time. As I claimed from the code each websocket connection is held in separate std::thread which is not acceptable for my project. Is it possible to hold 10 websocket connections in one thread with api of libhv ? As I understand there should be some epoll looping over 10 sockets and processing the input data.

**Comment by ithewei** [comment, marked as answer]
`WebSocketClient(EventLoopPtr loop = NULL);`
Consturctor can pass a EventLoopPtr.
refer to [examples/websocket_client_test.cpp](https://github.com/ithewei/libhv/blob/master/examples/websocket_client_test.cpp) TestMultiClientsRunInOneEventLoop

**Reply by HACKONDEX** [reply]
Thanks a lot. Now things are much more clear. It was strange for me because I used to create EventLoops passing threads count as constructor argument:)
